lib/prism/ripper_compat.rb

```diff
@@ -133,8 +167,22 @@ def visit_imaginary_node(node)
 
     # Visit an IntegerNode node.
     def visit_integer_node(node)
-      bounds(node.location)
-      on_int(node.slice)
+      text = node.slice
+      loc = node.location
+      if text[0] == "-"
+        # TODO: This is ugly. We test that a newline between - and 7 doesn't give an error, but this could still be bad somehow.
+        bounds_line_col(loc.start_line, loc.start_column + 1)
+        on_int_val = on_int(text[1..-1])
+        bounds(node.location)
+        if RUBY_ENGINE == "jruby"
+          on_unary(:-, on_int_val)
+        else
```

Comment:
I would suggest to run `test/prism/ripper_compat_test.rb` only on CRuby (`return unless RUBY_ENGINE == "ruby"`) and match only CRuby's behavior here.
Any difference of Ripper on JRuby/TruffleRuby is most likely an unintended incompatibility with CRuby.